File: Aio-Algorithms-And-Solutions/Solutions/HackerRank/GridChallenge/GridChallenge.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ConsoleTestSolutions.HackerRank.GridChallenge
{
    class GridChallenge
    {
        public static string gridChallenge(List<string> grid)
        {            
            for (int i = 0; i < grid.Count; i++)    
            {
                grid[i] = string.Concat(grid[i].OrderBy(ch => ch));    
            }
            
            for (int i = 0; i < grid[0].Length; i++)
            {
                for (int j = 0; j < grid.Count - 1; j++)
                {
                    if (grid[j].ToCharArray()[i] > grid[j + 1].ToCharArray()[i]) return "NO";  
                }
            }

            return "YES";
        }
    }
}

File: Aio-Algorithms-And-Solutions/Solutions/HackerRank/GridChallenge/GridChallenge.java
import java.util.Arrays;
import java.util.List;

public class GridChallenge {
    public static String gridChallenge(List<String> grid)
    {
        char[][] arr = new char[grid.size()][grid.size()];

        for (int i = 0; i < grid.size(); ++i)
        {
            arr[i] = grid.get(i).toCharArray();
            Arrays.sort(arr[i]);
        }

        for (int i = 0; i < arr[0].length; ++i)
        {
            for (int j = 0; j < arr.length - 1; ++j)
            {
                if (arr[j][i] > arr[j+1][i]) return "NO";
            }
        }
        return "YES";
    }
}